.github/workflows/python-package-conda.yml

```
@@ -3,6 +3,30 @@ name: Python Package using Conda
+  # Runs in seconds and needs no environment, so it sits in the workflow that triggers on
+  # every push rather than only on main-targeted PRs. Its absence is why poetry.lock drifted
+  # out of sync with pyproject.toml across ~28 commits without CI noticing: nothing in the
+  # pipeline installed via poetry or validated the lock, so the two could disagree forever
+  # while every job stayed green.
+  lockfile:
+    name: poetry.lock in sync
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v3
```

actionlint (1.7.12)

[error] the runner of "actions/setup-python@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)